sql/core/src/test/scala/org/apache/spark/sql/TPCDSCollationQueryTestSuite.scala

```diff
@@ -251,7 +251,10 @@ class TPCDSCollationQueryTestSuite extends QueryTest with TPCDSBase with SQLQuer
       test(name)(runQuery(queryString, Map.empty, emptyResults.contains(name)))
     }
 
-    tpcdsQueriesV2_7_0.foreach { name =>
+    // Skip q22-v2.7 in GitHub Action environment because it takes 14 minutes.
+    tpcdsQueriesV2_7_0
+      .filterNot(sys.env.contains("GITHUB_ACTIONS") && _.equals("q22"))
```

Comment:
Unlike `excludedTpcdsQueries`, we don't have `excludedTpcdsQueriesV2_7_0`. So, we need to filter out here.

Reply:
Should we leave a TODO to make sure we don't forget about this?

Reply:
Ya, I considered that too, but I didn't do that because this is not this query issue. This is a general `Collation`-related performance issue and I'm not sure who will take a look at that. 

However, let me add one for the community.